draft-ietf-ppm-dap.md

```diff
@@ -654,14 +655,27 @@ configuration of each aggregator. See {{compliance}} for information on HPKE
 algorithm choices.
 
 Clients retrieve the HPKE configuration from each aggregator by
-sending an HTTP GET request to `[aggregator]/hpke_config?task_id=[task-id]`, where
-`[aggregator]` is the aggregator's endpoint URL, obtained from the task
-parameters, and `[task-id]` is the task ID obtained from the task parameters,
+sending an HTTP GET request to `[aggregator]/hpke_config`, where `[aggregator]`
+is the aggregator's endpoint URL, obtained from the task parameters.
+
+A deployment MAY enforce different HPKE configurations for different tasks.
+In this case, clients MUST specify a query parameter `task_id` when
```

Comment:
I think MUST isn't right in this case, because it's not a violation of the protocol for the client to not send a `task_id` query parameter when the aggregator lacks a "global" HPKE config. Same goes for the one one line 669, because an aggregator is always free to respond to a request without the `task_id` query parameter.

Reply:
I think this paragraph tries to characterize the behaviors of an aggregator, when the deployment needs to enforce different HPKE configurations for different tasks. In this case, clients are required to send `task_id`, and aggregators are required to abort if no `task_id` is included in the request. Should I be clearer about the assumptions in this paragraph, or should I relax the "MUST" keyword, perhaps to "SHOULD"?

Reply:
I agree with Tim here. The problem is that the Client may not know a priori that the enforces HPKE key separation per task. I think a different tack is needed.

In the paragraph above where you specify the GET request, you should also add: "The Client MAY include the task ID with the query parameters. [Spell out what the path looks like.]"

In this paragraph, you should start off with: "The Aggregator is free to use different HPKE configurations for each task with which it's configured. If the task ID is missing from the Client's request, the Aggregator MAY abort with error "missingTaskID", in which case the Client SHOULD retry the request with the task ID included."

This accomplishes two things:
* It provides a way to obtain an HPKE config, regardless of how the Aggregator is configured.
* It permits the optional, deployment-specific beahvior.

Reply:
@tgeoghegan @cjpatton Please take a look again at the revised wording based on Chris's suggestions.

Reply:
Looks good to me.